### PR TITLE
wifi: UMAC Interface file clean up

### PR DESCRIPTION
**Pull request by srinivas1201**
1. Added doxygen style comments to missing commands and events in UMAC interface files
2. Removed unused commands and its corresponding structures in interface files
3. Updated command structures for NRF_WIFI_UMAC_CMD_GET_REG and NRF_WIFI_UMAC_CMD_START_P2P_DEVICE

### CODE REVIEW

**Review by krish2718 — CHANGES_REQUESTED** [review]
Please see https://github.com/nrfconnect/sdk-nrf/blob/main/drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fmac_structs_common.h#L24 for NRF style doxygent comments, we need to change for all structures.

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw/host_rpu_umac_if.h [hunk 1]
```diff
@@ -2495,6 +2495,17 @@ struct nrf_wifi_umac_cmd_get_tx_power {
...
 
+/**
+ * struct nrf_wifi_umac_cmd_get_reg - get regulatory.
+ * @umac_hdr: UMAC command header. Refer &struct nrf_wifi_umac_hdr.
+ *
+ * This structure represents the command to get regulatory domain.
+ */
+
+struct nrf_wifi_umac_cmd_get_reg {
+	struct nrf_wifi_umac_hdr umac_hdr;
```
Comment (krish2718) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw/host_rpu_umac_if.h:2506]: do we need need an empty structu? Can't we use just usee the `nrf_wifi_umac_hdr` ?

Comment (srinivas1201) [reply drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw/host_rpu_umac_if.h:2506]: Yes we can use. We have other command structures with only "umac_hdr", to align with them used above command.
we can further cleanup this by replacing all the empty structures with "umac_hdr", but that will show lot of differences from the interface file from UMAC side which went in to ROM.
I prefer to take up this change in vega.

**Review by rado17 — APPROVED** [review]

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw/host_rpu_common_if.h [hunk 2]
```diff
@@ -6,8 +6,8 @@
  */
 
 /**
+ *@brief Common interface between host and RPU
```
Comment (richabp) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw/host_rpu_common_if.h:9]:
```suggestion
 *@brief Common interface between host and RPU.
```

**Review by richabp — CHANGES_REQUESTED** [review]

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw/host_rpu_sys_if.h [hunk 3]
```diff
@@ -56,602 +56,772 @@
...
+ * @brief This structure specifies the configuration parameters used for configuring
+ *  data-related settings.
  *
- * Data configuration parameters provided in command NRF_WIFI_CMD_INIT
  */
 
 struct nrf_wifi_data_config_params {
+	/** rate_protection_type:0->NONE, 1->RTS/CTS, 2->CTS2SELF */
 	unsigned char rate_protection_type;
+	/** Aggreegation is enabled(NRF_WIFI_FEATURE_ENABLE) or
```
Comment (richabp) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw/host_rpu_sys_if.h:681]:
```suggestion
	/** Aggregation is enabled(NRF_WIFI_FEATURE_ENABLE) or
```

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw/host_rpu_sys_if.h [hunk 4]
```diff
@@ -664,385 +834,498 @@ struct nrf_wifi_cmd_sys_deinit {
...
 
+/**
+ * @brief The structure defines the parameters used to configure the coexistence hardware.
+ *
+ */
+
 struct rpu_cmd_coex_config_info {
+	/** Length of coexistence configuration data */
 	unsigned int len;
+	/** Coexistance configuration data */
```
Comment (richabp) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw/host_rpu_sys_if.h:1100]:
```suggestion
	/** Coexistence configuration data */
```

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw/host_rpu_sys_if.h [hunk 5]
```diff
@@ -664,385 +834,498 @@ struct nrf_wifi_cmd_sys_deinit {
...
 } __NRF_WIFI_PKD;
 
+/**
+ * @brief This structure defines the event used to represent coexistence configuration.
+ *
+ */
 struct nrf_wifi_event_coex_config {
+	/** UMAC header, @ref nrf_wifi_sys_head */
 	struct nrf_wifi_sys_head sys_head;
+	/** Coexistance configuration data in the event. @ref rpu_evnt_coex_config_info */
```
Comment (richabp) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fw/host_rpu_sys_if.h:1134]:
```suggestion
	/** Coexistence configuration data in the event. @ref rpu_evnt_coex_config_info */
```

**Review by richabp — APPROVED** [review]

**Pull request merged**